Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -252,6 +252,20 @@ if (AMReX_HIP)
    endif ()
    target_link_libraries(amrex PUBLIC hip::hiprand roc::rocrand roc::rocprim)
 
+
+   # add openmp library flag
+   if(AMReX_OMP)
```

Comment:
I don't think this new block is needed as we already have the capability in the build system a few lines earlier in this file :)
https://github.com/AMReX-Codes/amrex/blob/2ccb8166b22c853ffe1ae3049ca44ddc082b4236/Tools/CMake/AMReXParallelBackends.cmake#L27-L45

Can you please remove this block and try if configuring with
```console
cmake -S . -B build -DAMReX_OMP=ON -DAMReX_GPU_BACKEND=HIP -DAMReX_AMD_ARCH=gfx908
```
already works?

Reply:
Good catch! That block was from my original PR, which was less flexible than the block you linked, since it assumed gnu OpenMP libs. I was able to configure with `cmake -S . -B build -DAMReX_OMP=ON -DAMReX_GPU_BACKEND=HIP -DAMReX_AMD_ARCH=gfx908 -DCMAKE_C_COMPILER=$(which hipcc) -DCMAKE_CXX_COMPILER=$(which hipcc)` if I used `PrgEnv-gnu` (with both blocks or by removing either one).
Keeping my original block and removing the L27-45 block allows me to configure with `PrgEnv-amd` as well, but I think it's finding an older / default version of the libs.
In any case, I've removed the block I added, but it's worth noting that this build path requires an environment where FindOpenMP.cmake works.

Reply:
Correct, `FindOpenMP.cmake` is the unified interface we want to use (and hint to or fix if broken).